File: go/raindrops/raindrops.go
package raindrops

import (
    "strconv"
    "strings"
)

func Convert(number int) string {

    var sb strings.Builder

    if number % 3 == 0 {
        sb.WriteString("Pling")
    }

    if number % 5 == 0 {
        sb.WriteString("Plang")
    }

    if number % 7 == 0 {
        sb.WriteString("Plong")
    }

    if sb.Len() == 0 {
        return strconv.Itoa(number)
    }

    return sb.String()
}

File: typescript/raindrops/raindrops.ts
export default class RainDrops {

    convert(value: number): string {

        let answer = "";

        if (value % 3 == 0) { answer += "Pling"; }
        if (value % 5 == 0) { answer += "Plang"; }
        if (value % 7 == 0) { answer += "Plong"; }

        return answer.length == 0
            ? "" + value
            : answer;
    }
}
